api/controllers/EventController.js

```diff
@@ -39,15 +39,45 @@ var EventController = {
     },
     
     update: function (req, res) {
-        
+        var id = req.swagger.params.id.value;
+        var eventWithNewProperties = {};
+
+        if (req.body.title)         eventWithNewProperties.title = req.body.title;
+        if (req.body.description)   eventWithNewProperties.description = req.body.description;
+        if (req.body.purpose)       eventWithNewProperties.purpose = req.body.purpose;
+        if (req.body.minAge)        eventWithNewProperties.minAge = req.body.minAge;
+        if (req.body.maxAge)        eventWithNewProperties.maxAge = req.body.maxAge;
+        eventWithNewProperties.updatedAt = Date.now();
+
+        Event.findOneAndUpdate(id, eventWithNewProperties,function (err, _) {
+            if (err) return res.status(500).json(err.message);
+
+            res.status(204).send()
+        })
     },
     
     delete: function (req, res) {
         
     },
 
+    addOneAttendee: function (req, res) {
```

Comment:
Because they both update the event, `update` and `addOneAttendee` could be just one endpoint, which is an authorized api call. Something like:
```
if (user === event.author) {
    update()
} else {
    addUserToEvent() 
}
```